var/spack/repos/builtin/packages/wrf/package.py

```diff
@@ -258,6 +260,25 @@ def do_configure_fixup(self):
                             )
                         ofh.write(line)
 
+        if self.spec.satisfies("@4.2 %aocc"):
+            rename(
+                "./arch/configure.defaults",
+                "./arch/configure.defaults.bak",
+            )
+            with open("./arch/configure.defaults.bak", "rt") as ifh:
+                with open("./arch/configure.defaults", "wt") as ofh:
+                    for line in ifh:
+                        if line.startswith("DM_"):
+                            line = line.replace(
+                                "mpif90 -DMPI2_SUPPORT",
+                                self.spec['mpi'].mpifc + " -DMPI2_SUPPORT"
+                            )
+                            line = line.replace(
+                                "mpicc -DMPI2_SUPPORT",
+                                self.spec['mpi'].mpicc + " -DMPI2_SUPPORT"
+                            )
+                        ofh.write(line)
```

Comment:
Not necessarily to be done in this PR, but this method contains a fair amount of <kbd>Ctrl-C</kbd> + <kbd>Ctrl-V</kbd> that is best removed for maintainability.

Reply:
Also, just to double-check, the lines above this change refer to:
```
"./arch/configure_new.defaults",
"./arch/configure_new.defaults.bak",
```
with a `*_new` in the name of the file. Is this correct that here we use:
```
"./arch/configure.defaults",
"./arch/configure.defaults.bak",
```
? Is the change in naming related to the change in version of `wrf`? If so this is worth a comment because it may be easy to miss it during a refactor.

Reply:
you are right "the change in naming related to the change in version of wrf".

Reply:
Can you add an explicit comment pointing that out?

Reply:
Yeah, sure.
Because of this limitation from WRF, we had to duplicate same steps as you pointed here.

Reply:
@alalazo  Added required comments, please check the latest commit.